tests/tx/test_indexes3.py

```diff
@@ -85,26 +91,31 @@ def test_tips_index_initialization(self):
 
     @pytest.mark.flaky(max_runs=3, min_passes=1)
     def test_topological_iterators(self):
-        manager = self._build_randomized_blockchain()
-        tx_storage = manager.tx_storage
+        tx_storage = self.manager.tx_storage
 
         # XXX: sanity check that we've at least produced something
-        self.assertGreater(tx_storage.get_count_tx_blocks(), 3)
+        total_count = tx_storage.get_count_tx_blocks()
+        self.assertGreater(total_count, 3)
+
+        # XXX: sanity check that the children metadata is properly set (this is needed for one of the iterators)
+        for tx in tx_storage.get_all_transactions():
+            assert tx.hash is not None
+            for parent_tx in map(tx_storage.get_transaction, tx.parents):
+                self.assertIn(tx.hash, parent_tx.get_metadata().children)
 
         # test iterators, name is used to aid in assert messages
         iterators = [
             ('traditional', tx_storage._topological_sort()),
             ('fast', tx_storage._topological_fast()),
```

Comment:
Should we rename them to `_topological_sort_dfs`, `_topological_sort_timestamp_index`, and `_topological_sort_metadata`?

Reply:
That would be great. I tried to avoid calling the other two implementations a "topological **sort**" because no **sorting** is taking place, at least not on the timestamp indexes that has everything _already_ sorted, well, the second implementation also doesn't _really_ sort anything, it just walks taking into consideration the `timestamp` and because of the "parent's have strictly smaller timestamp" rule, the walk is naturally in topological order. But I think this is a small detail to fuss about with the naming, since they're still (and probably will for a while) considered internal, I think it's fine. I'll go with your suggestions since they give a much better idea of what the methods are doing.

Reply:
Great! Thanks! I agree that they are not really sorting but what matters is that the traversal order is a valid topological order.